src/background/modules/network/infrastructure.rs

```diff
@@ -29,10 +29,9 @@ pub fn register_network_events() -> Result<()> {
     if !REGISTERED.load(Ordering::Acquire) {
         REGISTERED.store(true, Ordering::Release);
         log::trace!("Registering network events");
-        NetworkManager::register_events(move |connectivity| {
+        NetworkManager::register_events(move |connectivity, ip| {
             log::trace!(target: "network", "Connectivity changed: {:?}", connectivity);
-            if let (Ok(ip), Ok(adapters)) = (get_local_ip_address(), NetworkManager::get_adapters())
-            {
+            if let (Ok(ip), Ok(adapters)) = (ip, NetworkManager::get_adapters()) {

```

Comment:
based on requested change, replace this line to: `if let Ok(adapters) = NetworkManager::get_adapters() {`